### ISSUE TITLE
Array copy behaviour

### ISSUE THREAD

**Issue by dkillick** [dask/dask#3751]
The following behaviour when copying a dask array created using `from_array` and then computing the result isn't what I'd expect:

```python
import dask.array as da
import numpy as np

d = da.from_array(np.array([0, 1]), chunks=2,)
dc = d.copy()

d is dc
False
d.compute() is dc.compute()
True
```

A dask array not created using `from_array` does not show this behaviour:

```python
d1 = da.ones(10, chunks=2,)

d1c = d1.copy()
d1 is d1c
False
d1.compute() is d1c.compute()
False
```

For reference, this behaviour seems to be tripping up [an Iris test](https://github.com/SciTools/iris/blob/master/lib/iris/tests/unit/cube/test_Cube.py#L1432-L1434), resulting in [this test failure](https://travis-ci.org/SciTools/iris/jobs/397436711#L5120-L5128). 

I'm using Python 3.6.5 and dask 0.18.1 (this behaviour doesn't appear in v0.18.0).

**Comment by jrbourbeau** [comment]
Hi @dkillick, thanks for opening up this issue! I believe the source of this difference is coming from a recent modification to `from_array` that added

https://github.com/dask/dask/blob/a95be8d16428671dda934f4c9217a6fc6023b89f/dask/array/core.py#L2331-L2333

in #3568. `d.compute() is dc.compute()` being `True` in your above example with dask 0.18.1 is because the `d` and `dc` only have a single chunk. For example, 

```python
import dask.array as da
import numpy as np

x = np.arange(10)
for name, chunks in zip(['single chunk', 'multiple chunks'], [-1, 2]):
    d = da.from_array(x, chunks=chunks)
    d_copy = d.copy()
    print(name)
    print('chunks = {}'.format(d.chunks))
    print(d is d_copy)
    print(d.compute() is d_copy.compute())
    print('\n')
```

should output something like

```
single chunk
chunks = ((10,),)
False
True

multiple chunks
chunks = ((2, 2, 2, 2, 2),)
False
False
```

One potential solution would be to update the `chunks` parameter in your test such that the dask array you're dealing with have more than a single chunk. Would this be a reasonable solution to fix your test failure?

**Comment by mrocklin** [comment]
We could also consider calling `self.map_blocks(np.copy)` if
self.npartitions == 1 within the copy function.

**Comment by dkillick** [comment]
Thanks @jrbourbeau @mrocklin! I'll explore setting multiple chunks in the test as a workaround while we await the release of the fix in #3754.
